File: actionbar/src/main/java/mobi/cangol/mobile/actionbar/internal/ActionMenuImpl.java
package mobi.cangol.mobile.actionbar.internal;

import android.view.View;

import java.util.ArrayList;
import java.util.List;

import mobi.cangol.mobile.actionbar.ActionMenu;
import mobi.cangol.mobile.actionbar.ActionMenuItem;
import mobi.cangol.mobile.actionbar.view.ActionMenuView;
import mobi.cangol.mobile.actionbar.view.ActionMenuView.OnActionClickListener;

/**
 * @author Cangol
 */
public final class ActionMenuImpl extends ActionMenu {
    private List<ActionMenuItem> mActions = new ArrayList<>();
    private ActionMenuView mActionMenuView;

    public ActionMenuImpl(ActionMenuView view) {
        this.mActionMenuView = view;
        this.mActionMenuView.setActionMenu(this);
    }

    @Override
    public List<ActionMenuItem> getActions() {
        return mActions;
    }

    @Override
    public void setActions(List<ActionMenuItem> actions) {
        if (actions != null){
            mActions=actions;
        }

        mActionMenuView.addActions(mActions);

    }

    @Override
    public ActionMenuItem addMenu(int id, int text, int drawable, int show) {
        ActionMenuItem item = new ActionMenuItem(id, text, drawable, show);
        mActions.add(item);
        mActionMenuView.addAction(item);
        return item;
    }

    @Override
    public void clear() {
        mActions.clear();
        mActionMenuView.removeAllActions();

    }

    @Override
    public int size() {
        return mActions.size();
    }

    @Override
    public ActionMenuItem getAction(int index) {
        return mActions.get(index);
    }

    @Override
    public void setOnActionClickListener(
            OnActionClickListener onActionClickListener) {
        mActionMenuView.setOnActionClickListener(onActionClickListener);

    }

    @Override
    public View getActionMenuItemView(int id) {
        return mActionMenuView.findViewById(id);
    }

}

File: actionbar/src/main/java/mobi/cangol/mobile/actionbar/internal/ActionMenuImpl.kt
package mobi.cangol.mobile.actionbar.internal

import android.view.View
import mobi.cangol.mobile.actionbar.ActionMenu
import mobi.cangol.mobile.actionbar.ActionMenuItem
import mobi.cangol.mobile.actionbar.ActionTabItem
import mobi.cangol.mobile.actionbar.view.ActionMenuView
import mobi.cangol.mobile.actionbar.view.ActionMenuView.OnActionClickListener

/**
 * @author Cangol
 */
class ActionMenuImpl(private val mActionMenuView: ActionMenuView) : ActionMenu() {
    private var mActions: MutableList<ActionMenuItem> = mutableListOf()

    init {
        this.mActionMenuView.setActionMenu(this)
    }

    override fun setActions(actions: MutableList<ActionMenuItem>) {
        mActions=actions
        mActionMenuView.addActions(mActions)
    }

    override fun getActions(): MutableList<ActionMenuItem> {
        return mActions
    }

    override fun addMenu(id: Int, text: Int, drawable: Int, show: Int): ActionMenuItem {
        val item = ActionMenuItem(id, text, drawable, show)
        mActions.add(item)
        mActionMenuView.addAction(item)
        return item
    }

    override fun clear() {
        mActions.clear()
        mActionMenuView.removeAllActions()

    }

    override fun size(): Int {
        return mActions.size
    }

    override fun getAction(index: Int): ActionMenuItem {
        return mActions[index]
    }

    override fun setOnActionClickListener(onActionClickListener: OnActionClickListener) {
        mActionMenuView.setOnActionClickListener(onActionClickListener)

    }

    override fun getActionMenuItemView(id: Int): View {
        return mActionMenuView.findViewById(id)
    }

}
